csrc/multidevice/communicator.h

```diff
@@ -108,6 +108,11 @@ class Communicator {
     return rankToDiD(rank_);
   }
 
+  // returns local rank
+  RankType local_rank() const {
+    return local_rank_;
```

Comment:
Do you mind code-comment `local_rank_`? E.g. how is it different from `rank_`? I suspect it's related to the concept of "node", but I failed to find what node means in this file either.

Reply:
Do you think "rank within a machine/node" more clear?